src/Storageless/Http/SessionMiddleware.php

```diff
@@ -115,33 +99,39 @@ public static function fromSymmetricKeyDefaults(string $symmetricKey, int $expir
      * This constructor simplifies instantiation when using HTTPS (REQUIRED!) and asymmetric key encryption
      * based on RSA keys
      */
-    public static function fromAsymmetricKeyDefaults(
-        string $privateRsaKey,
-        string $publicRsaKey,
+    public static function fromRsaAsymmetricKeyDefaults(
+        Signer\Key $privateRsaKey,
+        Signer\Key $publicRsaKey,
         int $expirationTime
     ): self {
         return new self(
-            new Signer\Rsa\Sha256(),
-            $privateRsaKey,
-            $publicRsaKey,
-            SetCookie::create(self::DEFAULT_COOKIE)
-                ->withSecure(true)
-                ->withHttpOnly(true)
-                ->withSameSite(SameSite::lax())
-                ->withPath('/'),
-            new Parser(),
+            Configuration::forAsymmetricSigner(
+                new Signer\Rsa\Sha256(),
+                $privateRsaKey,
+                $publicRsaKey
+            ),
+            self::buildDefaultCookie(),
             $expirationTime,
             new SystemClock(new DateTimeZone(date_default_timezone_get()))
         );
     }
 
+    public static function buildDefaultCookie(): SetCookie
```

Comment:
For those who construct `SessionMiddleware` by themselves, this will be pretty handy